### PR TITLE
New readme

### PR DESCRIPTION
**Pull request by samhumeau**
**Patch description**
Revamp of the Readme, clearer, shorter.

### CODE REVIEW

**Comment by jaseweston** [comment]
- i think the list of tasks could be better at the beginning, maybe saying how many there are at the beginning

- mention zoo at the beginning too? 

- and research/projects page?

- i think if we could replace s2s example with transformer could be nice, and not on babi

- the code structure (link) could be a link to ?:   https://github.com/facebookresearch/ParlAI/tree/new_readme/parlai

- the detailed goals also doesnt seem to fit now? i don't know .. it looks very wordy because nothing else is wordy

**Comment by emilydinan** [comment]
I think we should update the examples or take them out. @hadasah may have opinions after writing the quick start.

File: README.md [hunk 1]
```diff
@@ -4,326 +4,117 @@
...
 
-- motivates building new datasets that will go in the repository
+## Documentation
 
-## Properties
+ - [Basics: world, agents, teachers, action and observations](https://parl.ai/docs/tutorial_basic.html)
+ - [List of available tasks/datasets](https://parl.ai/docs/tasks.html)
+ - [Creating a dataset/task](http://www.parl.ai/docs/tutorial_task.html).
+ - [List of available agents](./parlai/agents)
+ - [Creating a new agent](https://parl.ai/docs/tutorial_seq2seq.html#).
```
Comment (hadasah) [inline README.md]: Nit: remove periods for this and the line two above.

File: README.md [hunk 2]
```diff
@@ -4,326 +4,117 @@
...
+This is the recommended installation procedure, as it provides ready access to the examples and allows you to modify anything you might need. This is especially useful if you if you want to submit another task to the repository.
 
-Set of datasets to bootstrap a working dialogue model for human interaction
+All needed data will be downloaded to ~/ParlAI/data, and any non-data files (such as the MemNN code) if requested will be downloaded to ~/ParlAI/downloads. If you need to clear out the space used by these files, you can safely delete these directories and any files needed will be downloaded again.
 
-- motivates building new datasets that will go in the repository
+## Documentation
 
-## Properties
+ - [Basics: world, agents, teachers, action and observations](https://parl.ai/docs/tutorial_basic.html)
```
Comment (hadasah) [inline README.md]: I think we probably want to add the Quick Start too.

File: README.md [hunk 3]
```diff
@@ -4,326 +4,117 @@
...
+- [**zoo**](./parlai/zoo): contains code to directly download and use pretrained models from our model zoo
 
-Our first release included the following datasets (shown in the left panel), and accessing one of them is as simple as specifying the name of the task as a command line option, as shown in the dataset display utility (right panel):
-<p align=center><img width="100%" src="docs/source/\_static/img/tasks2.png" /></p>
 
-Over 20 tasks were supported in the first release, including popular datasets such as
-SQuAD, bAbI tasks, MCTest, WikiQA, WebQuestions, SimpleQuestions, WikiMovies, QACNN, QADailyMail, CBT, BookTest, bAbI Dialog tasks,
-Ubuntu, OpenSubtitles, Cornell Movie, VQA-COCO2014.
-Since then, several datasets have been added such as  VQAv2, VisDial, MNIST_QA, Personalized Dialog, InsuranceQA, MS MARCO, TriviaQA, and CLEVR. See [here](https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/task_list.py) for the current complete task list.
+## Detailed Goals
```
Comment (hadasah) [inline README.md]: Yeah, having 3 or 4 separate bulleted lists under this is really awkward. I'm not sure we should have this section at all - but if we do, we should condense it better.

File: README.md [hunk 4]
```diff
@@ -4,326 +4,117 @@
...
-- Both fixed datasets (conversation logs) and interactive (online/RL) tasks.
-- Both real and simulated tasks.
-- Supports other media, e.g. visual in VQA.
-- Can use Mechanical Turk to run / collect data / evaluate.
-- Python framework.
-- Examples of training with PyTorch.
-- Supports batch and hogwild training and evaluation of models.
 
-## Basic Examples
+## Examples
```
Comment (hadasah) [inline README.md]: Yeah, we talked about having examples that are more similar to what users will likely actually use ParlAI for. Maybe we should just grab the display_data, train_model, and eval_model from the transformer on twitter section.

File: README.md [hunk 5]
```diff
@@ -4,326 +4,117 @@
...
+- many popular **datasets available all in one place, with the same API**, among them [SQuAD](https://rajpurkar.github.io/SQuAD-explorer/), [bAbI tasks](https://arxiv.org/abs/1502.05698), [MS MARCO](http://www.msmarco.org/), [WikiQA](https://www.microsoft.com/en-us/download/details.aspx?id=52419), [SimpleQuestions](https://arxiv.org/abs/1506.02075), [WikiMovies](https://arxiv.org/abs/1606.03126), [QACNN & QADailyMail](https://arxiv.org/abs/1506.03340), [CBT](https://arxiv.org/abs/1511.02301), [BookTest](https://arxiv.org/abs/1610.00956), [bAbI Dialogue tasks](https://arxiv.org/abs/1605.07683), [Ubuntu Dialogue](https://arxiv.org/abs/1506.08909), [OpenSubtitles](http://opus.lingfil.uu.se/OpenSubtitles.php), [Cornell Movie](https://www.cs.cornell.edu/~cristian/Cornell_Movie-Dialogs_Corpus.html), [VQA-COCO2014](http://visualqa.org/), [VisDial](https://arxiv.org/abs/1611.08669) and [CLEVR](http://cs.stanford.edu/people/jcjohns/clevr/) and more than 70 others. See the complete list [here](https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/task_list.py)
+- a wide set of **reference models** -- from retrieval baselines to Transformers.
```
Comment (pringshia) [inline README.md]: Is Transformers supposed to be capitalized as such?

File: README.md [hunk 6]
```diff
@@ -4,326 +4,117 @@
...
-End goal is real dialogue with people
+This will link the cloned directory to your site-packages.
 
-- train and evaluate on live dialogue with humans via Mechanical Turk or Messenger
-- easy setup for connecting turkers with your dialogue agent
-- allow to compare different research groups turk experiments
+This is the recommended installation procedure, as it provides ready access to the examples and allows you to modify anything you might need. This is especially useful if you if you want to submit another task to the repository.
 
-Set of datasets to bootstrap a working dialogue model for human interaction
+All needed data will be downloaded to ~/ParlAI/data, and any non-data files (such as the MemNN code) if requested will be downloaded to ~/ParlAI/downloads. If you need to clear out the space used by these files, you can safely delete these directories and any files needed will be downloaded again.
```
Comment (pringshia) [inline README.md]: What is the MemNN code? Perhaps this should be a link for reference? The qualification here is nice, but a way to get more context for beginners would be nice.

Comment (pringshia) [reply README.md]: I also find it easier to read when file paths are wrapped in code blocks, e.g. `~/ParlAI/downloads`... I notice this in some places like here:

https://github.com/facebookresearch/ParlAI/pull/1748/files#diff-04c6e90faac2675aa89e2176d2eec7d8R89

Maybe we can do it everywhere.

File: README.md [hunk 7]
```diff
@@ -4,326 +4,117 @@
...
-
-Included are examples of training neural models with [PyTorch](http://pytorch.org/), with batch training on GPU or hogwild training on CPUs. Using [Tensorflow](https://www.tensorflow.org/) or other frameworks instead is also straightforward.
-
-Our aim is for the number of tasks and agents that train on them to grow in a community-based way.
+- a **unified framework** for sharing, training and testing dialogue models from open-domain chitchat to VQA.
```
[the remark this answers is not in the document]
Comment (pringshia) [reply README.md]: This bullet point also seems a little repetitive given the first sentence, perhaps we could somehow merge this bullet right into the first sentence:

> ParlAI (pronounced “par-lay”) is a framework for dialogue AI research, implemented in Python.

into 

> ParlAI (pronounced “par-lay”) is a **unified framework** for dialogue AI research, implemented in Python. It provides a platform for sharing, training and testing dialogue models &mdash; from open-domain chitchat to VQA (visual question answering).

[the remark this answers is not in the document]
Comment (pringshia) [reply README.md]: Potential buzzword we can use: "[batteries](https://www.python.org/dev/peps/pep-0206/#batteries-included-philosophy)-[included](https://www.quora.com/What-does-batteries-included-philosophy-mean)"

File: README.md (hunk 3 above)
Comment (samhumeau) [reply README.md]: I removed that section, added a few stuff in the first paragraph

File: README.md (hunk 4 above)
Comment (samhumeau) [reply README.md]: This is a good idea, this is an actual good example

**Comment by samhumeau** [comment]
@jaseweston  :
"I'm not sure the "Basics: world, agents, teachers, action and observations" actually explains the way worlds and main loop works is explained as well as the original README, so we need some work there"
So Margaret have been remixing these in https://github.com/facebookresearch/ParlAI/pull/1731/files

**Comment by samhumeau** [comment]
Requesting another review @stephenroller  @jaseweston @hadasah @pringshia :

- removed detailed goals an property but added some stuff in the main list of gains in first part
- changed the examples for transformer ones
- nits and bits correction

**Comment by emilydinan** [comment]
@samhumeau  is this ready to merge

**Comment by samhumeau** [comment]
I think so I was waiting for opinions

**Pull request merged**